servers/rendering/renderer_viewport.cpp

```diff
@@ -823,34 +827,26 @@ void RendererViewport::viewport_set_scaling_3d_scale(RID p_viewport, float p_sca
 	_configure_3d_render_buffers(viewport);
 }
 
-uint32_t RendererViewport::Viewport::get_view_count() {
-	uint32_t view_count = 1;
-
-	if (use_xr && XRServer::get_singleton() != nullptr) {
-		Ref<XRInterface> xr_interface;
-
-		xr_interface = XRServer::get_singleton()->get_primary_interface();
-		if (xr_interface.is_valid()) {
-			view_count = xr_interface->get_view_count();
-		}
-	}
-
-	return view_count;
-}
-
 void RendererViewport::viewport_set_size(RID p_viewport, int p_width, int p_height) {
 	ERR_FAIL_COND(p_width < 0 && p_height < 0);
 
 	Viewport *viewport = viewport_owner.get_or_null(p_viewport);
 	ERR_FAIL_COND(!viewport);
 
-	viewport->size = Size2(p_width, p_height);
+	_viewport_set_size(viewport, p_width, p_height, viewport->view_count);
```

Comment:
I wonder if we should set view_count to 1 here, this would now always be called when the size is set on the Viewport node side which now only happens if use_xr is off. In this case the viewcount should return to 1 or it will remain 2 even if use_xr has been turned off.

Reply:
Yeah, that makes a lot of sense, because XR will never set the viewport size in this way. I'll update the PR when I'm back at my computer!

Reply:
I changed this to set the view count to 1, and also added a check to prevent it actually doing anything when use_xr is true, just in case someone tried to call it directly on the `RenderingServer`